examples/auto_ml_checkpoint.py

```diff
@@ -0,0 +1,133 @@
+import shutil
+import time
+
+import numpy as np
+
+from neuraxle.base import BaseStep, NonFittableMixin
+from neuraxle.checkpoints import DefaultCheckpoint
+from neuraxle.hyperparams.distributions import RandInt
+from neuraxle.hyperparams.space import HyperparameterSpace
+from neuraxle.metaopt.random import RandomSearch
+from neuraxle.pipeline import ResumablePipeline, DEFAULT_CACHE_FOLDER, Pipeline
+
+EXPECTED_OUTPUTS = np.array(range(10, 20))
+
+DATA_INPUTS = np.array(range(10))
+
+HYPERPARAMETER_SPACE = HyperparameterSpace({
+    'multiplication_2__hp_mul': RandInt(1, 3),
+    'multiplication_3__hp_mul': RandInt(1, 3),
+    'multiplication_4__hp_mul': RandInt(1, 3),
+    'multiplication_5__hp_mul': RandInt(1, 3),
+    'multiplication_6__hp_mul': RandInt(1, 3),
+    'multiplication_7__hp_mul': RandInt(1, 3),
+    'multiplication_8__hp_mul': RandInt(1, 3),
+    'multiplication_9__hp_mul': RandInt(1, 3),
+    'multiplication_10__hp_mul': RandInt(1, 3),
+    'multiplication_11__hp_mul': RandInt(1, 3)
+})
+
+
+class Multiplication(NonFittableMixin, BaseStep):
+    def __init__(self, sleep_time=0.1, hyperparams=None, hyperparams_space=None):
+        BaseStep.__init__(self, hyperparams=hyperparams, hyperparams_space=hyperparams_space)
+        self.sleep_time = sleep_time
+
+    def transform(self, data_inputs):
+        time.sleep(self.sleep_time)
+        if not isinstance(data_inputs, np.ndarray):
+            data_inputs = np.array(data_inputs)
+
+        return data_inputs * self.hyperparams['hp_mul']
+
+
+def main():
+    run_random_search_normal_pipeline()
+    run_random_search_resumable_pipeline()
+
+    shutil.rmtree(DEFAULT_CACHE_FOLDER)
```

Comment:
apply same review comments than for the notebook, pipeline.flush_cache or something.

Reply:
Note: we'll also want users to be sure to use a subfolder in the cache, for instance: 

`./cache/Pipeline/...`. 

Reason why I want to delete just the pipeline's self.name subfolder: If the user ever pass `./` instead of `./cache` as a cache folder, we should avoid deleting the whole cache folder, and just delete the `Pipeline/` subfolder of the cache, to avoid that the users do dangerous deletions or avoid to delete their whole projects at once lol.

Reply:
As a result, the `./cache` folder will be created, then the pipeline would use it. Then, on deleting the pipeline, The cache folder will be empty. EMpty files are not added to git by default so it should not be too bad to not delete the folder.